Fixes for NIRSpec

AarynnCarter opened the pull request:
For certain .ecf parameters it is not possible to perform a background subtraction (see #193) as the median of the error arrays is exactly 0. Adjusting the NIRSpec code to mask these zeroes when calculating the bg_err values. 

This likely only applies for the simulated data, so a warning has been added to the log to remind people when the real data comes around. 

Also removed the manual adjustment from NaN's to 0's as it is no longer needed.

AarynnCarter commented:
Just adjusted bright2flux.py to be able to digest the NIRSpec photom crds file. Also added a quick option to convert from MJy to DN/s within the existing `bright2dn()` function using the steradian to pixel conversion in the SCI header. 

This addresses Issue #169 partially, I could probably get NIRISS incorporated next week also.

taylorbell57 commented on eureka/S3_data_reduction/bright2flux.py:
```diff
@@ -118,6 +118,8 @@ def bright2dn(data, meta):
+    elif meta.inst == 'nirspec':
+        ind = np.where((foo['filter'] == data.mhdr['FILTER']) * (foo['grating'] == data.mhdr['GRATING']) * (foo['slit'] == data.shdr['SLTNAME']))[0][0] 
     else:
         raise ValueError(f'The bright2dn function has not been edited to handle the instrument {meta.inst}, and can currently only handle miri and nircam observations.')
```
You can add NIRSpec to the list in this message now

taylorbell57 commented on eureka/S3_data_reduction/s3_reduce.py:
```diff
@@ -173,6 +173,7 @@ def reduceJWST(eventlabel, ecf_path='./', s2_meta=None):
                 from . import nircam as inst
             elif meta.inst == 'nirspec':
                 from . import nirspec as inst
+                log.writelog('WARNING: Are you using real JWST data? Take a look at at the flag_bg() function in nirspec.py and Issue #193 on Github!')
```
I would instead write:
`log.writelog('WARNING: Are you using real JWST data? If so, you should edit the flag_bg() function in nirspec.py and look at Issue #193 on Github!')`

taylorbell57 commented:
Are there any related issues that can be closed with this PR?

AarynnCarter commented:
Okay I incorporated your tweaks @taylorbell57, and I've also added functionality to do the conversion for NIRISS as well.

However, @afeinstein20, @kevin218, I'd appreciate a few more sets of eyes on the NIRISS line, as I'm not entirely sure whether order=1 is the correct choice, or we need to do something more complicated. 

@taylorbell57 this closes #193, and #169, and maybe we could close #99 at the same time. However, I'm not sure how to formally connect these to the pull request beyond including them in the comment text.

taylorbell57 commented:
> @taylorbell57 this closes #193, and #169, and maybe we could close #99 at the same time. However, I'm not sure how to formally connect these to the pull request beyond including them in the comment text.

I added them for now, but in the future you can either use the Development section on the right of a PR or use "closing keywords" in your comments (described [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
